src/fairseq2/models/wav2vec2/model.py

```diff
@@ -125,7 +126,23 @@ def forward(self, batch: SequenceBatch) -> Wav2Vec2Output:
         # TODO: Should pad for fp16?
         encoder_output, _ = self.encoder(seqs, padding_mask)
 
-        return self.quantize_and_contrast(encoder_output, targets, temporal_mask)
+        return Wav2Vec2Features(
+            encoder_output=encoder_output, targets=targets, temporal_mask=temporal_mask
+        )
+
+    def forward(self, batch: SequenceBatch) -> Wav2Vec2Output:
+        """
+        :param batch:
+            The batch of sequences to process.
+        """
+        feats = self.extract_features(batch)
+
+        return self.quantize_and_contrast(
```

Comment:
@cbalioglu with the removal of `return_features`, this can just take the `Wav2Vec2Features` object directly, or keep the individual args as it was previously. Do you have a preference on this front?

Reply:
I think it makes sense to pass the `feats` object directly to `quantize_and_contrast`. It will make the code less noisy.